src/backend/drm/device/mod.rs

```diff
@@ -400,13 +407,23 @@ impl DrmDevice {
     }
 
     /// Actives a previously paused device.
-    pub fn activate(&self) {
+    ///
+    /// Specifying `true` for `disable_connectors` will call [`DrmDevice::reset_state`] if
+    /// the device was not active before. Otherwise you need to make sure there are no
+    /// conflicting requirements when enabling or creating surfaces.
+    pub fn activate(&mut self, disable_connectors: bool) -> Result<(), Error> {
         if self.device_fd().is_privileged() {
             if let Err(err) = self.acquire_master_lock() {
                 error!("Failed to acquire drm master again. Error: {}", err);
             }
         }
+        if disable_connectors && !self.is_active() {
+            self.reset_state()?;
```

Comment:
I'm guessing this shouldn't early-return before `set_active(true)`? Otherwise it's kinda weird because if you don't request resetting the state then you'll get the device marked as active regardless.

Reply:
I feel like, if the user specifically requests a reset and it fails, we shouldn't set it active, since the device likely disappeared or we lack permissions because it was moved between seats or something like that.

Reply:
But then if you activate(false) followed right away by reset_state(), then you will both get an error, and drm in active state

Reply:
True, this is what I mean with better documenting the function. By providing false you guarantee that you already brought the device into a known state that will work (or will do so before calling any other drm related function).

Reply:
But I get you point, it could be a surprising behavior and I should probably also gate `reset_state` begin `active`.
Doing so will make it impossible to call it before setting the device to active again.

Reply:
Okay, so this now always sets the device to active and won't allow you to call `reset_state` on a suspended device.
If you receive an error from reset_state (or acticate with true for disable_connectors) you are doomed in most cases anyway.